PubHub/PubHub.Common/Extensions/ServiceExtensions.cs

```diff
@@ -22,17 +23,31 @@ private static IServiceCollection AddApiService(this IServiceCollection services
             if (apiOptions == null)
                 throw new ArgumentNullException(nameof(services), "Options cannot be empty.");
 
+            services.AddScoped<Func<Task<TokenInfo>>>((sp) => () => apiOptions.TokenInfoAsync.Invoke(sp));
+
```

Comment:
What's this line for?
I don't understand what's going in 🤣

Reply:
I add a `Func` to access the currently stored tokens to the service provider from the another Func that used the service provider to get the tokens. Does that that makes sense? 😂 I thought it was quite clever 😅

Reply:
I'd call it arrogant :D 
(_But it's quite clever_)